rerun/src/main/java/org/apache/falcon/rerun/handler/RetryConsumer.java

```diff
@@ -43,14 +43,18 @@ public RetryConsumer(T handler) {
     protected void handleRerun(String clusterName, String jobStatus,
                                RetryEvent message, String entityType, String entityName) {
         try {
-            if (!jobStatus.equals("KILLED")) {
-                LOG.debug("Re-enqueing message in RetryHandler for workflow with same delay as job status is running:"
-                        + " {}", message.getWfId());
+            // Can happen when user does a manual re-run in-between retries.
+            if (jobStatus.equals("RUNNING") || jobStatus.equals("PREP")) {
+                LOG.debug("Re-enqueing message in RetryHandler for workflow with same delay as "
+                        + "job status is {} for : {}", jobStatus, message.getWfId());
                 message.setMsgInsertTime(System.currentTimeMillis());
                 handler.offerToQueue(message);
                 return;
+            } else if (jobStatus.equals("SUSPENDED") || jobStatus.equals("SUCCEEDED")) {
+                LOG.debug("Not retrying workflow {} anymore as it is in {} state. ", message.getWfId(), jobStatus);
```

Comment:
Makes sense. Just thinking if there could be a corner case where the instance after resuming had gone to timed out state or so because of which retry not happening.